app/routes/datasets/queries.py

```diff
@@ -552,6 +563,102 @@ async def _add_geometry_filter(parsed_sql, geometry: Geometry):
     return parsed_sql
 
 
+async def _add_revision_filter(parsed_sql, dataset, version):
+    """Add revision filters to geostore table queries.
+
+    For queries of latest dataset versions, this involves excluding rows from revisions
+    that are deleted by subsequent revisions in the version's revision history.
+
+    For queries of intermediate revisions, this involves including all rows from
+    append revisions and excluding rows from deleted revisions in the revision's
+    history.
+    """
+
+    default_asset: AssetORM = await assets.get_default_asset(dataset, version)
+
+    if default_asset.asset_type == AssetType.revision:
+        source_version = default_asset.revision_history[0]["version"]
+        source_default_asset: AssetORM = await assets.get_default_asset(
+            dataset, source_version
+        )
+        revision_asset = default_asset
+        latest_revision = source_default_asset.latest_revision
+
+    else:
+        latest_revision = default_asset.latest_revision
+        try:
+            revision_asset: AssetORM = await assets.get_default_asset(
+                dataset, latest_revision
+            )
+        except RecordNotFoundError:
+            revision_asset = None
+
+    sql_where = parsed_sql[0]["RawStmt"]["stmt"]["SelectStmt"].get("whereClause", None)
+
+    if revision_asset is None:
+        revision_filters = []
+    elif revision_asset.version == latest_revision:
+        revision_filters = await _filter_by_revision_operation(revision_asset, "latest")
+    else:
+        revision_filters = await _filter_by_revision_operation(
+            revision_asset, "intermediate"
+        )
+
+    if not revision_filters:
+        return parsed_sql
+
+    if sql_where:
+        parsed_sql[0]["RawStmt"]["stmt"]["SelectStmt"]["whereClause"] = {
+            "BoolExpr": {"boolop": 0, "args": [*sql_where, *revision_filters]}
+        }
+    else:
+        parsed_sql[0]["RawStmt"]["stmt"]["SelectStmt"]["whereClause"] = {
+            "BoolExpr": {"boolop": 0, "args": revision_filters}
+        }
+
+    return parsed_sql
+
+
+async def _filter_by_revision_operation(revision_asset: AssetORM, filter_type: str):
```

Comment:
What do the output filters look like for this? It seems like it'd be something like

`gfw_version != 'v1' AND gfw_version !='v2' AND ...`

If so, I think it'd make more sense to just make an IN statement, and making it like

`gfw_version NOT IN ('v1', v2', ...)`

I think this runs faster in SQL when there's a lot of filters.